### ISSUE TITLE
[ERROR] Node.js  setup_16.x | bash - on Debian Installation wrong version  18.x

### ISSUE THREAD

**Issue by liang611zh** [nodesource/distributions#1583]
**Describe your bug**
When install node in our pipline, we got a wrong version of node.

**Distribution Information:**
 - OS: [Debian]
 - script is in the [bitbucket-piplines.yml]
 - Images used:
    docker : php 8.1

**Node Version:**
 - Node: [Node.js v16.x:]

**To Reproduce**
`curl -sL https://deb.nodesource.com/setup_16.x | bash -`
`apt-get -y install nodejs`
`node -v`

**Expected behavior**
By using the same script.
Before Jun 15 our pipline will install correct version 16.x,
but yesterday this script will install a wrong version 18.x which caused the build process failed.

**Screenshots**
<img width="803" alt="Screen Shot 2023-06-16 at 13 23 58" src="https://github.com/nodesource/distributions/assets/16942462/b0977ee3-5d0d-4dbf-bc17-40e1ca289712">

**Comment by antocorso** [comment]
hello,
I think the problem is that the images php:8.1 (in your case and php:8.2 in my case) have been updated to debian 12 (that isn't in the supported list until now!).

**Comment by WhisperClaims** [comment]
We're having the same issue with our Ruby Node image. Node 18 is installed with breaking changes. Any suggested fix?

**Comment by liang611zh** [comment]
> hello, I think the problem is that the images php:8.1 (in your case and php:8.2 in my case) have been updated to debian 12 (that isn't in the supported list until now!). I solved my problem installing nvm choosing the node versione I needed (16)

Thank you antocorso, I switched to nvm source for node 16 it works.

But the issue in this case is still there.

**Comment by WhisperClaims** [comment]
You're a legend @liang611zh

**Comment by o1y** [comment]
This issue seems to be a duplicate of #1579, see https://github.com/nodesource/distributions/issues/1579#issuecomment-1594658955

tl;dr You have to pin `deb.nodesource.com` before installing the nodejs package. Otherwise it will use the default debian package sources, which contain node 18.x since Debian Bookworm, which was released on June 10th.

```bash
echo "Package: nodejs
Pin: origin deb.nodesource.com
Pin-Priority: 600" > /etc/apt/preferences.d/nodesource
```

**Comment by liang611zh** [comment]
> This issue seems to be a duplicate of #1579, see [#1579 (comment)](https://github.com/nodesource/distributions/issues/1579#issuecomment-1594658955)
> 
> tl;dr You have to pin `deb.nodesource.com` before installing the nodejs package. Otherwise it will use the default debian package sources, which contain node 18.x since Debian Bookworm, which was released on June 10th.
> 
> ```shell
> echo "Package: nodejs
> Pin: origin deb.nodesource.com
> Pin-Priority: 600" > /etc/apt/preferences.d/nodesource
> ```

Thank you o1y, I saw your issue before I open this issue. In my case the 'Pin' is not work, but I think this will help others who have same problem.

**Comment by o1y** [comment]
@liang611zh 

After adding the nodesource repository, can you check the output from `apt-cache policy nodejs`? Maybe you have to adjust the priority.

**Comment by liang611zh** [comment]
Looks we found solution to help people who have the same issue.
I think I can closed this issue. Thanks for every one.
